Tests/GRPCHTTP2CoreTests/GRPCStreamStateMachineTests.swift

```diff
@@ -1982,14 +1969,10 @@ final class GRPCStreamServerStateMachineTests: XCTestCase {
 
   func testReceiveMetadataWhenClientClosedAndServerClosed() {
     var stateMachine = self.makeServerStateMachine(targetState: .clientClosedServerClosed)
-
-    XCTAssertThrowsError(
-      ofType: RPCError.self,
-      try stateMachine.receive(headers: .clientInitialMetadata, endStream: false)
-    ) { error in
-      XCTAssertEqual(error.code, .internalError)
-      XCTAssertEqual(error.message, "Client can't have sent metadata if closed.")
-    }
+    XCTAssertEqual(
+      try stateMachine.receive(headers: .clientInitialMetadata, endStream: false),
+      .doNothing
```

Comment:
If the client is closed, should we be receiving data from them?

Reply:
It depends when you consider the "close" to happen. When the server sends a status it indicates that the RPC is done, so anything from the client at that point can be ignored (because the server doesn't need to process it). 

When the server sends the status I think it makes sense for the state machine to close _both_ sides, but I can also see the argument for the client remaining open until the end stream is received.

Reply:
WDYT?